components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementServiceImpl.java

```diff
@@ -98,8 +98,10 @@ public Device getCoreDevice(DeviceIdentifier deviceId) throws DeviceManagementEx
     }
 
     @Override
-    public boolean updateDeviceInfo(Device device) throws DeviceManagementException {
-        return DeviceManagementDataHolder.getInstance().getDeviceManagementProvider().updateDeviceInfo(device);
+    public boolean updateDeviceInfo(DeviceIdentifier deviceIdentifier, Device device) throws DeviceManagementException {
```

Comment:
In AndroidDeviceDAO, it is implemented to return true when a device successfully add, remove or update. Hence it is being used here as well to acknowledge.